### ISSUE TITLE
zcashd v1.0.8 crashed (out of memory)

### ISSUE THREAD

**Issue by ucwong** [zcash/zcash#2341]

**Comment by ucwong** [comment]
![image](https://cloud.githubusercontent.com/assets/22344498/25694022/fe01e5d6-30de-11e7-8dd4-342afc7427bd.png)

**Comment by radix42** [comment]
That looks like an out of memory error

**Comment by ucwong** [comment]
@radix42  never happened before.....

**Comment by radix42** [comment]
that error only is raised when a process can't allocate memory...how much ram do you have, and how much is free?

**Comment by ucwong** [comment]
@radix42 I think it is enough for zcashd
![image](https://cloud.githubusercontent.com/assets/22344498/25694231/3a4286c6-30e0-11e7-8cb0-1d2661a35d6e.png)
![image](https://cloud.githubusercontent.com/assets/22344498/25694236/42600e14-30e0-11e7-811a-92f5d14df087.png)

**Comment by radix42** [comment]
ok yeah that IS weird....lemme google deeper

**Comment by radix42** [comment]
ah its command will probably fix it: `sudo sysctl vm.overcommit_memory=1`
but that'll be lost at reboot and you'll need to add setting that your sysctl.conf
Try running that command and then firing zcashd up again

**Comment by ucwong** [comment]
@radix42 Thank you so much , I will try it .

**Comment by nathan-at-least** [comment]
Hi @ucwong, I re-opened this ticket, because I want to investigate why you ran out of memory. One immediate hypothesis is that the mempool of your node grew very large, due to recent network activity.

Here is a snapshot of one of our internal metric systems (which we hope to export to the public someday) showing recent mempool growth:

![screenshot 2017-05-04 at 14 35 25](https://cloud.githubusercontent.com/assets/4369700/25727266/29ebb048-30dd-11e7-9b78-d4e929ae8780.png)

One way to track if this was the cause of your memory issue would be to run `zcash-cli getmempoolinfo` periodically and watch the size.

**Comment by zapv0lt** [comment]
IMHO the mempool doesn't seem large enough to cause an OOM error, but I suppose it's possible.  @ucwong, another avenue of enquiry: do you use getblocktemplate on this node, e.g. for mining?

**Comment by zapv0lt** [comment]
(I'm only asking this because my zcashd was eventually OOM killed the other day; I suspect this is related to memory fragmentation over time when using getblocktemplate, in particular with the recent larger-than-usual mempool.  See my [comment](https://github.com/zcash/zcash/issues/2333#issuecomment-298342427) linking to a potentially-related Bitcoin issue which goes into more depth.)

**Comment by ucwong** [comment]
![image](https://cloud.githubusercontent.com/assets/22344498/25732954/531dc848-3187-11e7-9f9b-fdbddad2576f.png)
memory keeps going high for some hours , then crash at last

**Comment by ucwong** [comment]
@zapv0lt 
![image](https://cloud.githubusercontent.com/assets/22344498/25732997/bdad0b92-3187-11e7-8758-b0bd565ecc9e.png)
I don't call gbt very often. But the memory will keep going high to crash. It is not normal

**Comment by reppolice** [comment]
After a few days running on Ubuntu I had my first crash trying to do a T2T transfer, as soon as I got my op code. I can't believe I ran out of memory at 8GB RAM, but I don't really know. Retrying the command I did not get a crash, but nor do I seem to have a succesful transfer. Is there a mininum transfer size perhaps, or a fee required (for the otherwise free T2T)? And more importantly, can I troubleshoot that crash?
